core/src/main/java/org/bf2/srs/fleetmanager/execution/impl/tasks/ScheduleRegistryTask.java

```diff
@@ -13,12 +15,15 @@
 @ToString(callSuper = true)
 public class ScheduleRegistryTask extends AbstractTask {
 
-    private String registryId;
+    private AccountInfo accountInfo;
+    private RegistryData registryData;
```

Comment:
This field is not needed, store the registry in the database and just pass the ID since the task is already retrieving it:
```
Optional<RegistryData> registryOptional = storage.getRegistryById(task.getRegistryData().getId());
```

Reply:
The problem with that approach is if the two tasks are assigned to the same node you'll try to modify the same row twice using different objects in the same session, which is not allowed in Hibernate. I'll give this some thought to try to avoid modifying the task data.

Reply:
This should not happen AFAIK, but I might be wrong. The task should execute in a transaction anyway.